src/atomWithValidateExtended.ts

```diff
@@ -0,0 +1,32 @@
+import { atom } from 'jotai';
+import { atomWithDefault } from 'jotai/utils';
+
+const ACTIONS: string[] = ['TOUCHED', 'UNTOUCHED'];
+
+export const atomWithValidateExtended = (baseAtom: any) => {
+  const extended = atomWithDefault((get) => ({
+    touched: false,
+    ...get(baseAtom),
+  }));
+
+  const derv = atom(
+    (get) => get(extended),
+    (get, set, update) => {
+      if (typeof update === 'string' && ACTIONS.indexOf(update) > -1) {
+        const prev = get(extended);
+        set(extended, {
+          ...prev,
+          touched: update === 'TOUCHED',
+        });
+        return;
+      }
+      const prev = get(extended);
+      set(extended, {
+        ...prev,
+        value: update,
```

Comment:
**untouched** True if control has not lost focus yet.
**touched** True if control has lost focus.
**pristine** True if user has not interacted with the control yet.
**dirty** True if user has already interacted with the control.

This was my basis for the implementation

Reply:
So, I confused with `pristine`. I implemented `dirty` being "if changed", which is different from `!pristine`.

Reply:
yeah, if the above approach seems like a decent one, we can add the symbols for all four there and user's can use the same setter to set value and form states since none of them have to do anything with the actual change in value. 

other than maybe `isDirty` since that does mean that the control was interacted with

Reply:
From https://react-hook-form.com/api/useform/formstate, `isDirty` is false if it's the value is same as default. So, I assume if a user changes back to the default value, it's not dirty.

```js
value = 'default' // pristine=true, dirty=false
setValue('something') // pristine=false, dirty=true
setValue('default') // pristine=false, dirty=false
```

Reply:
From https://react-hook-form.com/api/useform/formstate, `touched` is "the user has interacted with." It's like `!pristine`. So confusing..

Reply:
tbh, whatever it's called, I doubt if we should have "focus" status in `atomWithValidate`. It can be a separate atom, which is more atomic and can be render optimized.

Reply:
okay, it looks like the terminology is very different between RHF and Formik. I don't mind following Formik terminology.

If we implement Formik's `touched`, I think it should run validation when touched becomes true. Then, it makes sense to have it in `atomWithValidate`.

Reply:
Yeah, it'd run for both touched and value change and before we finalize this approach, I'd like to find a way to keep the current atom as is and make `touched` to be an atomic state to the actual field state since there's also cases where you wouldn't want the required to be triggered unless `submit` is pressed. 

So, I'll add in a few more approaches and we can choose what seems like a reasonable and consistent API to all of jotai.